src/Mono.Android/Android.Runtime/JNIEnv.cs

```diff
@@ -193,60 +177,34 @@ internal static unsafe void Initialize (JnienvInitializeArgs* args)
 
 			Mono.SystemDependencyProvider.Initialize ();
 
-#if JAVA_INTEROP
+			// TODO: env, javaVm, grefLoader, Loader_loadClass should all be obtained by AndroidRuntime constructor with p/invokes
```

Comment:
Related: P/Invokes may be "fast", but let's not do one P/Invoke per value.  Bundle them!  Minimize P/Invoke transitions!

Reply:
That's the plan :)